### PR TITLE
fix: use new slot syntax instead of deprecated one

### PR DESCRIPTION
**Pull request by aantipov**
This PR addresses issues listed [here](https://github.com/vuejs/vuejs.org/issues/2058) + a wrong link fix

### CODE REVIEW

File: src/v2/api/index.md [hunk 1]
```diff
@@ -1385,23 +1385,25 @@ type: api
 
 - **Details:**
 
-  Used to programmatically access content [distributed by slots](../guide/components.html#Content-Distribution-with-Slots). Each [named slot](../guide/components.html#Named-Slots) has its own corresponding property (e.g. the contents of `slot="foo"` will be found at `vm.$slots.foo`). The `default` property contains any nodes not included in a named slot.
+  Used to programmatically access content [distributed by slots](../guide/components.html#Content-Distribution-with-Slots). Each [named slot](../guide/components.html#Named-Slots) has its own corresponding property (e.g. the contents of `v-slot:foo` will be found at `vm.$slots.foo`). The `default` property contains either nodes not included in a named slot or contents of `v-slot:default`.
+
+  **Note:** `vm.$slots` also supports slots defined via [deprecated syntax](../guide/components-slots.html#Deprecated-Syntax).
```
Comment (phanan) [inline src/v2/api/index.md]: I'd suggest something more specific e.g: 

> "v-slot:foo" is supported in v2.6+. For older versions, you can use the deprecated syntax…

File: src/v2/api/index.md [hunk 2]
```diff
@@ -1423,7 +1425,7 @@ type: api
   ```
 
 - **See also:**
-  - [`<slot>` Component](#slot-1)
+  - [`<slot>` Component](#slot)
```
Comment (phanan) [inline src/v2/api/index.md]: 🤗

File: src/v2/api/index.md (hunk 1 above)
Comment (aantipov) [reply src/v2/api/index.md]: Changed the note

**Comment by phanan** [comment]
Thanks!

**Pull request merged**